tests/publiccloud/create_aistack_env.pm

```diff
@@ -0,0 +1,247 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Basic aistack test
+
+# Summary: This test performs the following actions
+#  - Create a VM in EC2 using SLE-Micro-BYOS later than 6.0 version
+#  - Install the required dependencies to install the aistack helm chart
+#  - Test access to OpenWebUI and run integration tests with Ollama and MilvusDB
+# Maintainer: Yogalakshmi Arunachalam <yarunachalam@suse.com>
+#
+
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use serial_terminal;
+use publiccloud::utils;
+use publiccloud::ssh_interactive;
+use transactional;
+use containers::k8s;
+use strict;
+use warnings;
+use utils;
+use publiccloud::utils;
+use transactional qw(process_reboot trup_install trup_shell);
+use File::Basename;
+use version_utils;
+use Data::Dumper;
+
+sub install_dependency_package {
+    my ($instance) = @_;
+    my $rke2_url = get_var('RKE2_URL');
+    my $kubectl_url = get_var('KUBECTL_URL');
+    my $helm_url = get_var('HELM_URL');
+
+    record_info('Dep pkg install');
+    trup_call("pkg install curl git docker");
+    process_reboot(trigger => 1);
+    systemctl("enable docker");
+    systemctl("start docker");
+    systemctl("status docker");
+    script_run("curl -sSL $rke2_url -o ./install_rke2.sh && chmod 775 ./install_rke2.sh");
+    script_run("sh ./install_rke2.sh");
+    script_run("echo 'export PATH=\$PATH:/opt/rke2/bin' >> ~/.bashrc");
+    systemctl("enable rke2-server.service");
+    systemctl("start rke2-server.service");
+    systemctl("status rke2-server.service");
+    script_run("rke2 --version");
+    script_run("curl -sSL $helm_url -o ./install_helm.sh && chmod 775 ./install_helm.sh");
+    script_run("sh ./install_helm.sh");
+    script_run("helm version");
+    script_run("curl -sSL $kubectl_url -o ./kubectl && chmod +x ./kubectl");
+    script_run("sudo mv ./kubectl /usr/local/bin/");
+    script_run("kubectl version --client");
+}
+
+sub install_dependency_components {
+    my ($instance) = @_;
+    my $cert_repo = get_var('HELM_CERTS');
+    my $ingress_repo = get_var('HELM_INGRESS');
+    my $ing_ver = get_var('ING_VERSION');
+    # Add Ingress Controller to open-webui endpoint
+    assert_script_run("helm repo add $ingress_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm upgrade --install ingress-nginx ingress-nginx/ingress-nginx --namespace ingress-nginx --set controller.service.type=ClusterIP --version $ing_ver --create-namespace", timeout => 120);
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --version v1.15.2 --set crds.enabled=true", timeout => 120);
+    sleep 180;
```

Comment:
I won't block on the sleeps, but perhaps there's a message in the logs you can use, to ensure that the installation didn't fail or didn't take too long.